packages/streaming-image-volume-loader/src/StreamingImageVolume.ts

```diff
@@ -412,10 +452,19 @@ export default class StreamingImageVolume extends ImageVolume {
       }
     }
 
-    const requests = imageIds.map((imageId, imageIdIndex) => {
+    // 4D datasets load one time point at a time and the frameIndex is
+    // the position of the imageId in the current time point while the
+    // imageIdIndex is its absolute position in the array that contains
+    // all other imageIds. In a 4D dataset the frameIndex can also be
+    // calculated as `imageIdIndex % numFrames` where numFrames is the
+    // number of frames per time point. The frameIndex and imageIdIndex
+    // will be the same when working with 3D datasets.
+    const requests = imageIds.map((imageId, frameIndex) => {
+      const imageIdIndex = this.imageIds.indexOf(imageId);
+
       if (cachedFrames[imageIdIndex]) {
-        framesLoaded++;
-        framesProcessed++;
+        this.framesLoaded++;
+        this.framesProcessed++;
         return;
       }
```

Comment:
same here, we don't need to explain what 4d does in the base class. This should be inside the 4d